Source/Fuse.Nodes/Visual.Children.uno

```diff
@@ -18,35 +18,59 @@ namespace Fuse
 	*/
 	public partial class Visual
 	{
-		public bool HasChildren { get { return _children.Count > 0; } }
+		public bool HasChildren { get { return ChildCount > 0; } }
 
 		protected override void SubtreeToString(StringBuilder sb, int indent)
 		{
 			base.SubtreeToString(sb, indent);
-			for (int i = 0; i < Children.Count; i++)
-				Children[i].SubtreeToString(sb, indent+1);
+			for (var c = FirstChild<Node>(); c != null; c = c.NextSibling<Node>())
+				c.SubtreeToString(sb, indent+1);
 		}
 
+		/** Returns the first child node of the given type. 
+			
+			To get the very first child node (of any type), use `FirstChild<Node>()`.
+		*/
 		public T FirstChild<T>() where T: Node
 		{
-			for (var i = 0; i < Children.Count; ++i)
+			var c = _firstChild;
+			while (c != null)
 			{
-				var c = Children[i] as T;
-				if (c != null)
-					return c;
+				var v = c as T;
+				if (v != null) return v;
+				c = c._nextSibling;
 			}
 			return null;
 		}
 
-		public void RemoveAllChildren<T>()
+		/** Returns the last child node of the given type. 
+
+			To get the very last child node (of any type), use `LastChild<Node>()`.
+		*/
+		public T LastChild<T>() where T: Node
 		{
-			var i = Children.Count - 1;
-			while (i >= 0)
+			var c = _lastChild;
+			while (c != null)
 			{
-				if (Children[i] is T)
-					Children.RemoveAt(i);
+				var v = c as T;
+				if (v != null) return v;
+				c = c._previousSibling; 
+			}
+			return null;
+		}
 
-				i = Math.Min(i, Children.Count) - 1;
```

Comment:
The reason the loop iterating backwards before, and did this check, was because removing one node could potentially cause additional nodes to be unrooted -- such as unrooting a behaviour that added children...